velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -74,10 +78,83 @@ class ParquetWriterTest : public ParquetTestBase {
             std::make_shared<InMemoryReadFile>(std::move(data)),
             opts.memoryPool()),
         opts);
-  };
+  }
+
+  MemorySink* write(
+      const RowTypePtr& schema,
+      const RowVectorPtr& data,
+      parquet::WriterOptions& writerOptions) {
+    auto sink = std::make_unique<MemorySink>(
+        200 * 1024 * 1024, FileSink::Options{.pool = leafPool_.get()});
+    auto* sinkPtr = sink.get();
+    auto writer = std::make_unique<parquet::Writer>(
+        std::move(sink), writerOptions, schema);
+    writer->write(data);
+    writer->close();
+    writers_.push_back(std::move(writer));
+    return sinkPtr;
+  }
+
+  MemorySink* write(
+      const RowTypePtr& schema,
+      const RowVectorPtr& data,
+      std::unordered_map<std::string, std::string> configFromFile = {},
+      std::unordered_map<std::string, std::string> sessionProperties = {}) {
+    parquet::WriterOptions writerOptions;
+    writerOptions.memoryPool = rootPool_.get();
+    auto connectorConfig = config::ConfigBase(std::move(configFromFile));
+    auto connectorSessionProperties =
+        config::ConfigBase(std::move(sessionProperties));
+    writerOptions.processConfigs(connectorConfig, connectorSessionProperties);
+    return write(schema, data, writerOptions);
+  }
+
+  std::pair<RowTypePtr, RowVectorPtr> makeSmallintTestData(int64_t rows) {
```

Comment:
we can just return RowVectorPtr; the caller can fetch RowTypePtr using RowVectorPtr::rowType()